sentry_sdk/integrations/aiohttp.py

```diff
@@ -164,6 +170,74 @@ async def sentry_urldispatcher_resolve(self, request):
 
         UrlDispatcher.resolve = sentry_urldispatcher_resolve
 
+        old_client_session_init = ClientSession.__init__
+
+        def init(*args, **kwargs):
+            # type: (Any, Any) -> ClientSession
+            hub = Hub.current
+            if hub.get_integration(AioHttpIntegration) is None:
+                return old_client_session_init(*args, **kwargs)
+
+            client_trace_configs = list(kwargs.get("trace_configs", ()))
+            trace_config = create_trace_config()
+            client_trace_configs.append(trace_config)
+
+            kwargs["trace_configs"] = client_trace_configs
+            return old_client_session_init(*args, **kwargs)
+
+        ClientSession.__init__ = init
+
+
+def create_trace_config():
+    # type: () -> TraceConfig
+    async def on_request_start(session, trace_config_ctx, params):
+        # type: (ClientSession, SimpleNamespace, TraceRequestStartParams) -> None
+        hub = Hub.current
+        if hub.get_integration(AioHttpIntegration) is None:
+            return
+
+        method = params.method.upper()
+
+        parsed_url = None
+        with capture_internal_exceptions():
+            parsed_url = parse_url(str(params.url), sanitize=False)
+
+        span = hub.start_span(
+            op=OP.HTTP_CLIENT, description="%s %s" % (method, parsed_url)
+        )
+        span.set_data(SPANDATA.HTTP_METHOD, method)
+        span.set_data("url", parsed_url.url)
+        span.set_data(SPANDATA.HTTP_QUERY, parsed_url.query)
+        span.set_data(SPANDATA.HTTP_FRAGMENT, parsed_url.fragment)
+
+        if should_propagate_trace(hub, parsed_url.url):
```

Comment:
```suggestion
        if should_propagate_trace(hub, str(params.url)):
```
This should still use the original unparsed URL. `parsed_url.url` can be altered by `parse_url` (don't get fooled by the `sanitize=False`, it still does [some rudimentary sanitization](https://github.com/getsentry/sentry-python/blob/0ebb2f93f7cd9990c987b56d0488613703749ef8/sentry_sdk/utils.py#L1410-L1412) 🙈 ).